pkgs/applications/science/electronics/qucs-s/default.nix

```
@@ -1,28 +1,26 @@
-{ stdenv, lib, fetchFromGitHub, flex, bison, qt4, libX11, cmake, gperf, adms,
+{ mkDerivation, lib, fetchFromGitHub, flex, bison, qtbase, qttools, qtscript, libX11, cmake, gperf, adms,
 ngspice, wrapGAppsHook,
 kernels ? [ ngspice ] }:
 
-stdenv.mkDerivation rec {
+mkDerivation rec {
   pname = "qucs-s";
-  version = "0.0.22";
+  version = "0.0.23";
 
   src = fetchFromGitHub {
     owner = "ra3xdh";
     repo = "qucs_s";
     rev = version;
-    sha256 = "0rrq2ddridc09m6fixdmbngn42xmv8cmdf6r8zzn2s98fqib5qd6";
+    hash = "sha256-EvRL3L/vD+H/rRuwhzAm+EvH4vsn42J9js/3p+vmeP0=";
   };
 
   nativeBuildInputs = [ wrapGAppsHook cmake ];
-  buildInputs = [ flex bison qt4 libX11 gperf adms ] ++ kernels;
+  buildInputs = [ flex bison qtbase qttools qtscript libX11 gperf adms ] ++ kernels;
```

```suggestion
  buildInputs = kernels;
```
1. `flex bison libX11 gperf adms` are not actually needed
2. `qttools` should be moved to `nativeBuildInputs`
3. `qtscript` is not needed ([qucs-s >= 0.0.24](https://github.com/ra3xdh/qucs_s/releases/tag/0.0.24))